client/web/auth.go

```diff
@@ -232,3 +233,48 @@ func (s *Server) newSessionID() (string, error) {
 	}
 	return "", errors.New("too many collisions generating new session; please refresh page")
 }
+
+type peerCapabilities map[capFeature]bool // value is true if the peer can edit the given feature
+
+// canEdit is true if the peerCapabilities grant edit access
+// to the given feature.
+func (p peerCapabilities) canEdit(feature capFeature) bool {
+	if p == nil {
+		return false
+	}
+	return p[feature]
+}
+
+type capFeature string
+
+const (
+	capFeatureAll       capFeature = "*"         // grants peer management of all features
+	capFeatureFunnel    capFeature = "funnel"    // grants peer serve/funnel management
+	capFeaturetSSH      capFeature = "ssh"       // grants peer SSH server management
+	capFeatureSubnets   capFeature = "subnets"   // grants peer subnet routes management
+	capFeatureExitNodes capFeature = "exitnodes" // grants peer advertising/using exit nodes
```

Comment:
Singular instead?

Reply:
Do you think the cap values themselves should also be singular?

Reply:
Not sure, it's your call.